src/LinAlg/hiopMatrixRajaDense.cpp

```diff
@@ -333,16 +331,17 @@ void hiopMatrixRajaDense::copyRowsFrom(const hiopMatrix& src_mat, const index_ty
   assert(n_rows<=src.m_local_);
   assert(n_rows == m_local_);
 
-  // todo //! opt -> copy multiple (consecutive rows at the time -> maybe keep blocks of eq and ineq,
-  //instead of indexes)
-
-  //int i should suffice for dense matrices
-  auto& rm = umpire::ResourceManager::getInstance();
   RAJA::View<double, RAJA::Layout<2>> Mview(this->data_dev_, m_local_, n_local_);
   RAJA::View<const double, RAJA::Layout<2>> Sview(src.data_dev_, src.m_local_, src.n_local_);
+  
   for(int i=0; i<n_rows; ++i)
   {
-    rm.copy(&Mview(i, 0), const_cast<double*>(&Sview(rows_idxs[i], 0)), n_local_ * sizeof(double));
+    //rm.copy(&Mview(i, 0), const_cast<double*>(&Sview(rows_idxs[i], 0)), n_local_ * sizeof(double));
+    RAJA::forall<hiop_raja_exec>(RAJA::RangeSegment(0, n_local_),
+    RAJA_LAMBDA(RAJA::Index_type j)
+    {
+      Mview(i, j) = Sview(rows_idxs[i], j);
+    });
```

Comment:
Not sure why the original one line approach doesn't work here.
Should we record all the patches like this in a file, e.g., READE_developers.md, or offline in a shared drive?
I think this will be useful when other developer meets a bug in hiop-RAJA-cuda code. Otherwise we may forget this after couple of months.

Reply:
because `rows_idxs` is on the device and the previous/original copy (based on Umpire's `copy`) accesses `rows_idxs` on the host (which ends in a crash). Not sure we need to document this as it falls under "common sense" imho